developer_documentation/rcdgen_markup.rst

```diff
@@ -0,0 +1,1096 @@
+:Author: The FreeRCT Team
+:Version: 2022-09-02
+
+.. contents::
+   :depth: 3
+
+#####################################
+FreeRCT Markup Language Specification
+#####################################
+
+.. Section levels  # = ~ .
+
+License
+=======
+This file is part of FreeRCT.
+FreeRCT is free software; you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, version 2.
+FreeRCT is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.
+See the GNU General Public License for more details. You should have received a
+copy of the GNU General Public License along with FreeRCT. If not, see
+<http://www.gnu.org/licenses/>.
+
+Introduction
+============
+This file documents the file format used by FreeRCT's *RCDgen* program to generate RCD files. See "data_format.rst" for more information about RCD files.
+
+RCDgen accepts input files in the **FreeRCT Preprocessed Markup Language (FPP)**.
+These files are generated by the C preprocessor out of
+**FreeRCT Markup Language (FML)** files and **FreeRCT Markup Snippet (FMS)** files.
+
+The three file formats differ only in that FPP files may not contain `preprocessor directives`_ except `line directives`_.
+
+
+Recommended File System Layout
+==============================
+The following section is a recommendation on how to layout the files. It is permitted to ignore it.
+
+An FML file is a top-level file. It must not be included in any other file.
+It should generate exactly one RCD file by providing exactly one top-level value list which is a `file`_ node.
+Except for the filename extension, the name of the generated file should be identical to the FML file's name.
+
+An FMS file is not preprocessed by itself. It is always included from another FMS or FML file.
+Each FMS file is included only one file.
```

Comment:
```suggestion
Each FMS file is included in only one other file.
```
Alternatively "Each FMS file is included only once." although it may be a bit less clear.